pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -519,18 +545,30 @@ func buildWaypointConnectOriginateListener(push *model.PushContext, proxy *model
 	return buildConnectOriginateListener(push, proxy, istionetworking.ListenerClassSidecarInbound)
 }
 
-func buildConnectOriginateListener(push *model.PushContext, proxy *model.Proxy, class istionetworking.ListenerClass) *listener.Listener {
+func buildWaypointForwardInnerConnectListener(push *model.PushContext, proxy *model.Proxy) *listener.Listener {
+	return buildForwardInnerConnectListener(push, proxy, istionetworking.ListenerClassSidecarInbound)
+}
+
+func buildForwardInnerConnectListener(push *model.PushContext, proxy *model.Proxy, class istionetworking.ListenerClass) *listener.Listener {
+	return buildConnectForwarder(push, proxy, class, ForwardInnerConnect, false)
+}
+
+func buildConnectForwarder(push *model.PushContext, proxy *model.Proxy, class istionetworking.ListenerClass,
+	clusterName string, tunnel bool,
+) *listener.Listener {
 	tcpProxy := &tcp.TcpProxy{
-		StatPrefix:       ConnectOriginate,
-		ClusterSpecifier: &tcp.TcpProxy_Cluster{Cluster: ConnectOriginate},
-		TunnelingConfig: &tcp.TcpProxy_TunnelingConfig{
+		StatPrefix:       clusterName,
+		ClusterSpecifier: &tcp.TcpProxy_Cluster{Cluster: clusterName},
+	}
+	if tunnel {
+		tcpProxy.TunnelingConfig = &tcp.TcpProxy_TunnelingConfig{
 			Hostname: "%DOWNSTREAM_LOCAL_ADDRESS%",
-		},
+		}
 	}
 	// Set access logs. These are filtered down to only connection establishment errors, to avoid double logs in most cases.
 	accessLogBuilder.setHboneOriginationAccessLog(push, proxy, tcpProxy, class)
```

Comment:
This seems wrong if we are not actually doing HBONE.

Reply:
Good catch